### PR TITLE
Remove dependency on xml-apis-xerces on the printing extension

### PR DESCRIPTION
**Pull request by groldan**
`xml-apis:xml-apis-xerces:2.7.1` is the same as
`xml-apis:xml-apis` but from 2012 and unnecessary since Java 1.6.

  

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/extension/printing/pom.xml [hunk 1]
```diff
@@ -35,11 +35,6 @@
...
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>xml-apis</groupId>
-      <artifactId>xml-apis-xerces</artifactId>
-      <version>2.7.1</version>
-    </dependency>
     <dependency>
       <groupId>xerces</groupId>
       <artifactId>xercesImpl</artifactId>
```
Comment (mprins) [inline src/extension/printing/pom.xml:40]: xercesImpl could probably be removed as well

Comment (groldan) [reply src/extension/printing/pom.xml:40]: I don't know enough about the module as to make an informed decision

**Comment by aaime** [comment]
@jodygarnett I'd feel better if you could have a look at this one.

**Comment by jodygarnett** [comment]
Understood

**Comment by jodygarnett** [comment]
@groldan how does the interact with mapfish-print-v2 release?

**Comment by groldan** [comment]
@jodygarnett your question made me realize I didn't remove the  `<include>xml-apis-xerces-*.jar</include>` include from `src/release/ext-printing.xml`. Just did.

That said, I'm not sure I understand your question.
The `xml-apis` dependency is not necessary since Java 1.6. The `xml-apis-xerces` dependency contains exactly the same classes as `xml-apis:xml-apis`. `xml-apis-xerces` is just released a year before `xml-apis:xml-apis`, and they don't even include the StAX apis.
So they're just ancient and all their classes have been included in the standard JRE system library for a long time (`javax.xml`, `org.s3c.dom`, and `org.xml.sax` namespaces).

**Comment by jodygarnett** [comment]
Thanks I was trying to ask if any similar change is useful for mapfish-print-v2 (since as you say xerces dependency not useful)

**Comment by aaime** [comment]
@groldan @jodygarnett I don't understand if this should be closed and moved upstream to mapfish-print-v2, or merged here...

**Comment by jodygarnett** [comment]
@groldan  can I remove xerces dependency from mapfish print?

**Pull request merged**